core/src/main/scala/org/apache/spark/deploy/worker/DriverWrapper.scala

```diff
@@ -23,14 +23,15 @@ import org.apache.commons.lang3.StringUtils
 
 import org.apache.spark.{SecurityManager, SparkConf}
 import org.apache.spark.deploy.{DependencyUtils, SparkHadoopUtil, SparkSubmit}
+import org.apache.spark.internal.Logging
 import org.apache.spark.rpc.RpcEnv
 import org.apache.spark.util.{ChildFirstURLClassLoader, MutableURLClassLoader, Utils}
 
 /**
  * Utility object for launching driver programs such that they share fate with the Worker process.
  * This is used in standalone cluster mode only.
  */
-object DriverWrapper {
+object DriverWrapper extends Logging {
```

Comment:
This `DriverWrapper` actually runs within the same JVM of driver, and initialize log4j instance earlier. Will this be a problem?

Reply:
I can not conceive, why that can be a problem. Can you also describe, why do you think that it can be a problem?